scrapers_next/ak/people.py

```diff
@@ -1,69 +1,143 @@
+            ]
+            details[district_phone_pattern] = district_phone
+
+        party_formatting = {
+            "Democrat": "Democratic",
+            "Republican": "Republican",
+            "Not": "Independent",
+        }
+        listed_party = details["Party"]
+        party = party_formatting[listed_party]
+
+        image = CSS(".legpic").match_one(self.root).get("src")
+
+        p = ScrapePerson(
+            name=f"{given_name} {family_name}",
+            given_name=given_name,
+            family_name=family_name,
             state="ak",
+            chamber=self.input.chamber,
             party=party,
-            chamber=("upper" if chamber == "S" else "lower"),
-            district=item_dict["District"],
-            image=f"http://akleg.gov/images/legislators/{code}.jpg",
-            email=item_dict["EMail"],
+            image=image,
+            district=details["District"],
+            email=email,
         )
-        person.add_link(
-            "http://www.akleg.gov/basis/Member/Detail/{}?code={}".format(
-                self.session_num, code
+
+        try:
+            leadership_title = (
+                CSS(".leadership_title").match(details_div)[0].text_content()
             )
+            p.extras["title"] = leadership_title
+        except SelectorError:
+            pass
+
+        if details.get("Phone"):
+            p.capitol_office.voice = details["Phone"]
+
+        if details.get(district_phone_pattern):
+            p.district_office.voice = details[district_phone_pattern]
+
+        if details.get("Fax"):
+            p.district_office.fax = details["Fax"]
+
+        if details.get(address_patterns[0]):
+            p.district_office.address = details[address_patterns[0]]
+        p.district_office.name = "interim contact"
+
+        if details.get(address_patterns[1]):
+            p.capitol_office.address = details[address_patterns[1]]
+        p.capitol_office.name = "session contact"
+
+        if details.get("Toll-Free"):
+            p.extras["toll_free_phone"] = details["Toll-Free"]
+
+        source_url = str(self.source)
+        p.add_source(source_url, "member detail page")
+
+        session_match = re.search(r"Detail/\d+", source_url)
+        session = source_url[session_match.start() + 7 : session_match.end()]
+        p.add_source(
+            f"https://www.akleg.gov/basis/mbr_info.asp?session={session}",
+            "member list page",
         )
-        person.add_source("http://w3.akleg.gov/")
 
-        if item_dict["Phone"]:
-            phone = "907-" + item_dict["Phone"][0:3] + "-" + item_dict["Phone"][3:]
-            person.capitol_office.voice = phone
+        return p
 
-        if item_dict["Building"] == "CAPITOL":
-            person.capitol_office.address = (
-                "State Capitol Room {}; Juneau, AK, 99801".format(item_dict["Room"])
-            )
 
-        return person
+class LegList(HtmlListPage):
+    session_num = "32"
```

Comment:
Nice following the practice of breaking this up even if it's set in the file instead of a passed in variable.

Reply:
Thanks!